modules/machine-health-checks-resource.adoc

```diff
@@ -39,9 +40,67 @@ format. For example, `prod-node-us-east-1a`.
 <5> Specify the amount of unhealthy machines allowed in the targeted pool of
 machines. This can be set as a percentage or an integer.
+<6> Specify how long a MachineHealthCheck should wait for a Node to join the cluster,
+before considering a Machine unhealthy. This should be a duration string.
 
 [NOTE]
 ====
 The `matchLabels` are examples only; you must map your machine groups based on
 your specific needs.
 ====
+
+== Remediation short-circuiting
+
+To ensure that MachineHealthChecks only remediate Machines when the cluster is healthy,
+short-circuiting is implemented to prevent further remediation via the `maxUnhealthy`
+field within the MachineHealthCheck.
+
+If the user defines a value for the `maxUnhealthy` field
+(either an absolute number or a percentage of the total Machines checked by this MachineHealthCheck),
+before remediating any Machines, the MachineHealthCheck will compare the value of `maxUnhealthy`
+with the number of Machines within its target pool that it has determined to be unhealthy.
+If the number of unhealthy Machines exceeds the limit set by `maxUnhealthy`, remediation will *not* be performed.
+
+[IMPORTANT]
+====
+If `maxUnhealthy` is unset, it is assumed to mean `100%`.
+This means the short circuiting mechanism is *disabled by default* and Machines
+will be remediated no matter the state of the cluster.
+====
+
+=== Examples
+
+==== With an absolute value
+
+If `maxUnhealthy` is set to `2`:
+- If 2 or fewer nodes are unhealthy, remediation will be performed
+- If 3 or more nodes are unhealthy, remediation will not be performed
+
+These values are independent of how many Machines are being checked by the MachineHealthCheck.
+
+==== With percentages
+
+If `maxUnhealthy` is set to `40%` and there are 25 Machines being checked:
+- If 10 or fewer nodes are unhealthy, remediation will be performed
+- If 11 or more nodes are unhealthy, remediation will not be performed
+
+If `maxUnhealthy` is set to `40%` and there are 6 Machines being checked:
+- If 2 or fewer nodes are unhealthy, remediation will be performed
+- If 3 or more nodes are unhealthy, remediation will not be performed
+
+Note, when the percentage is not a whole number, the allowed number is rounded down.
+
+== Limitations and Caveats
+
+Before deploying a MachineHealthCheck, please familiarise yourself with the
+following limitations and caveats:
+
+- Only Machines owned by a MachineSet will be remediated by a MachineHealthCheck.
+- Control Plane Machines are currently not supported and will *not* be remediated
+  if they are unhealthy.
+- If the Node for a Machine is removed from the cluster, a MachineHealthCheck will
+  consider this Machine unhealthy and remediate it immediately.
+- If no Node joins the cluster for a Machine after the `NodeStartupTimeout`,
+  the Machine will be remediated.
+- If a Machine fails for any reason (if the phase is `Failed`),
+  the Machine will be remediated immediately.
```

Comment:
Just want to check on this one, is it normal to say `node` when referring to a Kubernetes Node resource, rather than capitalising it as we have for the other Kubernetes resources?